### PR TITLE
Lk1552 

### PR DESCRIPTION
**Pull request by reish981**

### CODE REVIEW

**Comment by kawesome** [comment]
Can one of the admins verify this patch?

File: src/main/resources/static/js/jquery.confirm.js [hunk 1]
```diff
@@ -144,4 +144,75 @@
...
+                if (value) {
+                    dataOptions[optionName] = value;
+                }
+            });
+        }
+
+        // Default options
+        var settings = $.extend({}, $.confirm.options, {
+            confirm: function () {
+                var url = e && (('string' === typeof e && e) || (e.currentTarget && e.currentTarget.attributes['href'].value));
```
Comment (KSU-Github) [inline src/main/resources/static/js/jquery.confirm.js]: CRITICAL Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=javascript%3AS1067)

**Comment by KSU-Github** [comment]
SonarQube analysis reported 9 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 8 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [jquery.confirm.js#L71](https://github.com/kstateome/lti-attendance/blob/78858bb08766d7b7382f8da7af516986bce32bbf/src/main/resources/static/js/jquery.confirm.js#L71): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=javascript%3AS1067)
1. ![CRITICAL][CRITICAL] [jquery.confirm.js#L82](https://github.com/kstateome/lti-attendance/blob/78858bb08766d7b7382f8da7af516986bce32bbf/src/main/resources/static/js/jquery.confirm.js#L82): Add a nested comment explaining why this function is empty or complete the implementation. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=javascript%3AS1186)
1. ![MAJOR][MAJOR] [jquery.confirm.js#L82](https://github.com/kstateome/lti-attendance/blob/78858bb08766d7b7382f8da7af516986bce32bbf/src/main/resources/static/js/jquery.confirm.js#L82): Remove the unused function parameter "o". [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [courseConfiguration.jsp#L9](https://github.com/kstateome/lti-attendance/blob/78858bb08766d7b7382f8da7af516986bce32bbf/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L9): Add a 'favicon' declaration in this 'header' tag. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3APageWithoutFaviconCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L109](https://github.com/kstateome/lti-attendance/blob/78858bb08766d7b7382f8da7af516986bce32bbf/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L109): Add a <legend> tag to this fieldset. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3AFieldsetWithoutLegendCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L141](https://github.com/kstateome/lti-attendance/blob/78858bb08766d7b7382f8da7af516986bce32bbf/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L141): Replace this <i> tag by <em>. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3ABoldAndItalicTagsCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L146](https://github.com/kstateome/lti-attendance/blob/78858bb08766d7b7382f8da7af516986bce32bbf/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L146): Replace this <i> tag by <em>. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3ABoldAndItalicTagsCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L152](https://github.com/kstateome/lti-attendance/blob/78858bb08766d7b7382f8da7af516986bce32bbf/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L152): Replace this <i> tag by <em>. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3ABoldAndItalicTagsCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by KSU-Github** [comment]
SonarQube analysis reported 7 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [jquery.confirm.js#L71](https://github.com/kstateome/lti-attendance/blob/616035ba49d461ac4fbfc418564ca5bb1cf1fab0/src/main/resources/static/js/jquery.confirm.js#L71): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=javascript%3AS1067)
1. ![MAJOR][MAJOR] [courseConfiguration.jsp#L9](https://github.com/kstateome/lti-attendance/blob/616035ba49d461ac4fbfc418564ca5bb1cf1fab0/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L9): Add a 'favicon' declaration in this 'header' tag. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3APageWithoutFaviconCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L109](https://github.com/kstateome/lti-attendance/blob/616035ba49d461ac4fbfc418564ca5bb1cf1fab0/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L109): Add a <legend> tag to this fieldset. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3AFieldsetWithoutLegendCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L141](https://github.com/kstateome/lti-attendance/blob/616035ba49d461ac4fbfc418564ca5bb1cf1fab0/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L141): Replace this <i> tag by <em>. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3ABoldAndItalicTagsCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L146](https://github.com/kstateome/lti-attendance/blob/616035ba49d461ac4fbfc418564ca5bb1cf1fab0/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L146): Replace this <i> tag by <em>. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3ABoldAndItalicTagsCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L152](https://github.com/kstateome/lti-attendance/blob/616035ba49d461ac4fbfc418564ca5bb1cf1fab0/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L152): Replace this <i> tag by <em>. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3ABoldAndItalicTagsCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by killsto** [comment]
In general it's probably better to define html needed for javascript manipulation in the jsp instead of in the javascript. It has a few benefits such as IDE's recognizing it and no surprises when editing the html. For a small example, if you had to add a class to a lot of elements the one defined in javascript would probably be missed. 

So, in this case you could create the modal on the jsp page and hide the div by default.

There are probably places in our codebases that don't do this so I can see being confused.

In cases where you are just appending things to some elements then html generation is probably fine.

**Comment by KSU-Github** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [jquery.confirm.js#L71](https://github.com/kstateome/lti-attendance/blob/dabf7da90537659bb21f48fa4dc0da86cd34c1fc/src/main/resources/static/js/jquery.confirm.js#L71): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=javascript%3AS1067)
1. ![MAJOR][MAJOR] [courseConfiguration.jsp#L9](https://github.com/kstateome/lti-attendance/blob/dabf7da90537659bb21f48fa4dc0da86cd34c1fc/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L9): Add a 'favicon' declaration in this 'header' tag. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3APageWithoutFaviconCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L136](https://github.com/kstateome/lti-attendance/blob/dabf7da90537659bb21f48fa4dc0da86cd34c1fc/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L136): Add a <legend> tag to this fieldset. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3AFieldsetWithoutLegendCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L168](https://github.com/kstateome/lti-attendance/blob/dabf7da90537659bb21f48fa4dc0da86cd34c1fc/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L168): Replace this <i> tag by <em>. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3ABoldAndItalicTagsCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L173](https://github.com/kstateome/lti-attendance/blob/dabf7da90537659bb21f48fa4dc0da86cd34c1fc/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L173): Replace this <i> tag by <em>. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3ABoldAndItalicTagsCheck)
1. ![MINOR][MINOR] [courseConfiguration.jsp#L179](https://github.com/kstateome/lti-attendance/blob/dabf7da90537659bb21f48fa4dc0da86cd34c1fc/src/main/webapp/WEB-INF/jsp/courseConfiguration.jsp#L179): Replace this <i> tag by <em>. [rule](https://sonar.ome.ksu.edu/coding_rules#rule_key=Web%3ABoldAndItalicTagsCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by reish981** [comment]
Don't merge this. I've found an issue with it. Doesn't seem to be updating grades in the gradebook or properly pushing to Canvas.

**Pull request closed without merging**